br/pkg/lightning/mydump/loader.go

```diff
@@ -297,7 +312,7 @@ func (s *mdLoaderSetup) setup(ctx context.Context, store storage.ExternalStorage
 	*/
 	var gerr error
 	if err := s.listFiles(ctx, store); err != nil {
-		if errors.ErrorEqual(err, common.ErrTooManySourceFiles) {
+		if s.setupCfg.ReturnPartialResultOnError {
```

Comment:
should this need to change too ?
https://github.com/pingcap/tidb/blob/5209ac2fa1a3db278661588370786b2991728320/br/pkg/lightning/restore/precheck.go#L76

Reply:
You are right.  Will push the patch soon

Reply:
fixed in 2e935901b98ea5379ff13a9e4dd02ad2b41486c9